management/src/main/scala/akka/management/AkkaManagementSettings.scala

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.management
+
+import java.net.InetAddress
+
+import com.typesafe.config.Config
+import scala.collection.JavaConverters._
+
+final class AkkaManagementSettings(val config: Config) {
+  private val managementConfig = config.getConfig("akka.management")
+
+  object Http {
+    private val cc = managementConfig.getConfig("http")
+
+    val Hostname: String = {
+      val hostname = cc.getString("hostname")
+      if (hostname == "") InetAddress.getLocalHost.getHostAddress
+      else hostname
+    }
+
+    val Port: Int = {
+      val p = cc.getInt("port")
+      require(p >= 0, s"akka.management.http.port MUST be > 0 (was ${p})")
+      p
+    }
+
+    def isHttps = false // FIXME
+
+    val BasePath: Option[String] =
+      Option(cc.getString("base-path")).flatMap(it ⇒ if (it.trim == "") None else Some(it))
+
+    val RouteProviders: List[String] = cc.getStringList("route-providers").asScala.toList
```

Comment:
This mechanism is the same as the "autostarting extensions" in Akka, each lib contributes to this list.
I really think it makes most sense like this since that's just "include lib and got more things in endpoint", and the active things, like the Bootstrap's `start()` are still required to be called by the user.

Same with binding the routes actually -- users have to call that because they may want to configure HTTPS etc

Reply:
I like this -- we can use it to provide, for instance, sane readiness checks on k8s and dcos based on cluster status.

Reply:
immutable.Seq instead of List